### PR TITLE
[Rust] Some rust cleanups

### PR DESCRIPTION
**Pull request by binarybana**
* Turn off default features for bindgen
* Upgrade some deps for smaller total dep tree
* Switch (/complete switch) to thiserror
* Remove unnecessary transmutes

Thanks for contributing to TVM!   Please refer to guideline https://tvm.apache.org/docs/contribute/ for useful information and tips. After the pull request is submitted, please request code reviews from [Reviewers](https://github.com/apache/incubator-tvm/blob/master/CONTRIBUTORS.md#reviewers) by @ them in the pull request thread.

### CODE REVIEW

File: rust/tvm-sys/Cargo.toml [hunk 1]
```diff
@@ -32,4 +32,4 @@ ndarray = "0.12"
 enumn = "^0.1"
 
 [build-dependencies]
-bindgen = "0.51"
+bindgen = { version="0.51", default-features=false }
```
Comment (jroesch) [inline rust/tvm-sys/Cargo.toml:35]: I think @kazum wanted this on for building the WASM backend, can you confirm?

**Comment by jroesch** [comment]
LGTM, @mwillsey can you review as well?

**Comment by binarybana** [comment]
I noticed a small build error I introduced in the last rebase. Will fix and push when I get a second, but wanted to flag it first.

File: rust/tvm-sys/Cargo.toml (hunk 1 above)
Comment (binarybana) [reply rust/tvm-sys/Cargo.toml:35]: The default-features of `bindgen` pulls in `clap` which has a pretty big (albeit `build-dep`) dependency load. So it'd be good to trim if possible.

**Comment by binarybana** [comment]
Okay, builds and tests pass now.

File: rust/tvm-sys/Cargo.toml (hunk 1 above)
Comment (jroesch) [reply rust/tvm-sys/Cargo.toml:35]: Thanks for following up @kazum all the big changes to Rust should be done. I will be sending a bunch of doc updates in coming weeks after I finish landing new Relay error reporting. Max and I have been using the bindings at OctoML for some experiments.

**Comment by jroesch** [comment]
Thanks Jason!

**Pull request merged**